### PR TITLE
Add Refresh.HttpsProxy

### PR DESCRIPTION
**Pull request by Beyley**
This adds a simple proxy capable of forwarding LBP3 traffic on ports 10061 (HTTPS) and 10060 (HTTP) to another gameserver, it also overrides the original server's digest, allowing clients with non-patched digest keys to connect without the actual gameserver having to support the proxied game's digest.

### CODE REVIEW

File: Refresh.HttpsProxy/Middlewares/ProxyMiddleware.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            queryString.Append('?');
+
+        // Fill in all the query strings
+        for (int i = 0; i < context.Query.Count; i++)
+        {
+            string key = context.Query.GetKey(i)!;
+            
+            string name = HttpUtility.HtmlEncode(key);
+
+            string value = HttpUtility.HtmlEncode(context.Query[key]!);
```
Comment (jvyden) [inline Refresh.HttpsProxy/Middlewares/ProxyMiddleware.cs:36]:
```suggestion
            string key = context.Query.GetKey(i)!;
            string name = HttpUtility.HtmlEncode(key);
            string value = HttpUtility.HtmlEncode(context.Query[key]!);
```

**Review by jvyden — CHANGES_REQUESTED** [review]
Couple small things

Will you be moving the digest changes over to the current middleware implementation at all or are you just keeping the refactor to the new one?

**Comment by Beyley** [comment]
> Will you be moving the digest changes over to the current middleware implementation at all or are you just keeping the refactor to the new one?

I will do that in a future PR

**Review by jvyden — APPROVED** [review]

**Comment by jvyden** [comment]
docker support would be helpful but i'll address that later

**Pull request merged**